packages/aws-cdk-lib/aws-s3/lib/notifications-resource/notifications-resource-handler.ts

```diff
@@ -84,13 +84,9 @@ export class NotificationsResourceHandler extends Construct {
 
     const handlerSource = fs.readFileSync(path.join(__dirname, '..', '..', '..', 'custom-resource-handlers', 'dist', 'aws-s3', 'notifications-resource-handler', 'index.py'), 'utf8');
 
-    // Removing lines that starts with '#' (comment lines) in order to fit the 4096 limit
+    // Removing lines that starts with '#' (comment lines)
     const handlerSourceWithoutComments = handlerSource.replace(/^ *#.*\n?/gm, '');
 
-    if (handlerSourceWithoutComments.length > 4096) {
```

Comment:
Inline function code can now exceed 4096 bytes. It can support up to 4MB now.

Latest public documentation: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-lambda-function-code.html#cfn-lambda-function-code-zipfile